TestTools/StreamChatTestMockServer/MockServer/MessageList.swift

```diff
@@ -18,6 +18,17 @@ public extension StreamMockServer {
         }
     }
     
+    func saveReply(_ message: [String: Any]?) {
+        guard let newMessage = message else { return }
+        if let index = threadList.firstIndex(where: { (message) -> Bool in
+            (newMessage[messageKey.id.rawValue] as? String) == (message[messageKey.id.rawValue] as? String)
+        }) {
+            threadList[index] = newMessage
+        } else {
+            threadList.append(newMessage)
+        }
+    }
```

Comment:
This function looks a bit confusing and hard to read, would my suggestion of putting the parent Id of the message, make it easier?

Reply:
Hmm, not sure I got you. I did update it a little bit (as well as `saveMessage()` as they are similar). Could you please check it out?